src/routes/forgot-password/components/ForgotPassword.js

```diff
@@ -0,0 +1,110 @@
+const React = require('react');
+const T = require('prop-types');
+const StrangeForms = require('strange-forms');
+const IsEmail = require('utils/is-email');
+
+module.exports = class ForgotPassword extends StrangeForms(React.Component) {
+
+    static propTypes = {
+        requestReset: T.func.isRequired,
+        errorMessage: T.string
+    };
+
+    constructor(props) {
+
+        super(props);
+
+        this.state = {
+            email: '',
+            isBlurred: {
+                email: false
+            }
+        };
+
+        this.strangeForm({
+            fields: ['email'],
+            get: (someProps, field) => this.state[field],
+            act: this.setFormField.bind(this),
+            getFormValue: this.getFormValue.bind(this)
+        });
+    }
+
+    getFormValue(e) {
+
+        return e.target.value || '';
+    }
+
+    setFormField(field, value) {
+
+        this.setState({ [field]: value });
+    }
+
+    fieldBlurred = (ev) => {
+
+        const isBlurred = { ...this.state.isBlurred };
+        const field = ev.target.id;
+
+        isBlurred[field] = true;
+
+        this.setState({ isBlurred });
+    }
+
+    showEmailError = () => {
+
+        return (this.state.isBlurred.email) && !IsEmail(this.state.email);
+    }
+
+    disableButton = () => {
+
+        const { email } = this.state;
+        const fieldHasValue = email !== '';
+
+        if (fieldHasValue && IsEmail(email)) {
```

Comment:
I think this could be simplified to 

```
return !IsEmail(email);
```

assuming `IsEmail` returns `false` for `''`.

This could just be me and my weird brain, but I felt slow following the logic here.  I think it's because I'm used to thinking about it as what validates the form, instead of when to disable the button (and similarly, in the disableButton function, I'd have the if statement define what makes it true instead of false).  I find it more straightforward to think about what criteria deem a valid form, and to negate that when determining if the button should be active.  Not tryin to say it's bad, just that it's different from my thought process.

Reply:
similar feedback in forms below

Reply:
thanks, I simplified this one! 

Yeah, I'm definitely thinking about this as _when should the button be disabled_ more than _is this form valid_ (and to your other point, I guess it make sense to me to have the default return true because the the exception would be a disabled button, meaning the button will spend more time being disabled (true) than enabled (false)).

I'll get more opinions on this approach!

Reply:
@devinivy would love to get your opinion (this convo got buried) !

Reply:
Is this primarily a question about how it should behave, or how it should be implemented?

Reply:
My original comment?  Implementation, I think.  I think the original code functions correctly (disables the button when it should), but I found myself needing to think...some when trying to follow the logic, so I threw out a potential simplification and general thoughts.

Reply:
Ah!  Yeah, I guess I don't have a strong opinion!  This is more stylistic, but in the case of a function that returns a boolean based upon a single expression, I would probably just return that expression.
```js
    // When should the form be disabled?
    // When the field doesn't have a value or isn't an email!
    return !fieldHasValue || !IsEmail(email);
```